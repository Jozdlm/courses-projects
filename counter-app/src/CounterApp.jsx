import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    setCounter((current) => current + 1);
  };

  const handleSubs = () => {
    if (counter > 0) {
      setCounter(counter - 1);
    }
  };

  const handleReset = () => {
    setCounter(value);
  };

  return (
    <div className="counter-wrapper">
      <h1>Counter App</h1>
      <p>{counter}</p>
      <button aria-label="button-substract" onClick={handleSubs}>
        -1
      </button>
      <button aria-label="button-reset" onClick={handleReset}>
        Reset
      </button>
      <button aria-label="button-add" onClick={handleAdd}>
        +1
      </button>
    </div>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
