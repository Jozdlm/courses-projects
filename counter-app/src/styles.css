* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  background-color: #21232a;
  color: white;
  font-size: 1rem;
  font-family: system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto,
    Oxygen, Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
}

.counter-wrapper {
  margin: 64px auto;
  max-width: 1000px;
  width: 90%;
}

button {
  padding: 15px;
  font-size: 1.3rem;
}
