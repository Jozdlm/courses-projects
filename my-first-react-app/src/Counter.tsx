import { ComponentProps, useState } from "react";

interface props extends ComponentProps<"div"> {
  value: number;
}

export const Counter = ({ value }: props) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    setCounter((value) => value + 1);
  };

  const handleMin = () => {
    setCounter((value) => {
      if (value > 0) {
        return value - 1;
      }

      return value;
    });
  };

  const handleReset = () => {
    setCounter(value);
  };

  return (
    <>
      <h2>{counter}</h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleMin}>-1</button>
    </>
  );
};
