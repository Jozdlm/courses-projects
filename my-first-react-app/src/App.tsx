import { Counter } from "./Counter";

export const App = () => {
  return (
    <>
      <h1>Counter App</h1>
      <Counter value={1} />
    </>
  );
};
