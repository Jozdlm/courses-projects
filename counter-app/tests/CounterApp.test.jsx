import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe("Testing in CounterApp component", () => {
  const value = 100;

  test("Should the component match with the snapshot", () => {
    const { container } = render(<CounterApp value={value} />);
    expect(container).toMatchSnapshot();
  });

  test("Should render the initial value as 100", () => {
    render(<CounterApp value={value} />);
    expect(screen.getByText(value)).toBeTruthy();
  });

  test("Should the value be incremented by one when the user click +1", () => {
    const initialValue = 10;

    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText("+1"));
    expect(screen.getByText(initialValue + 1)).toBeTruthy();
  });

  test("Should the value be decremented by one when the user click -1", () => {
    const initialValue = 10;

    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText("-1"));
    expect(screen.getByText(initialValue - 1)).toBeTruthy();
  });

  test("Should the value be equals to the initial value when the user click reset", () => {
    const initialValue = 10;

    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByRole("button", { name: "button-reset" }));
    expect(screen.getByText(initialValue)).toBeTruthy();
  });
});
