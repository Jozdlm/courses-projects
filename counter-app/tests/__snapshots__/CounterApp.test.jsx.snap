// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in CounterApp component Should the component match with the snapshot 1`] = `
<div>
  <div
    class="counter-wrapper"
  >
    <h1>
      Counter App
    </h1>
    <p>
      100
    </p>
    <button
      aria-label="button-substract"
    >
      -1
    </button>
    <button
      aria-label="button-reset"
    >
      Reset
    </button>
    <button
      aria-label="button-add"
    >
      +1
    </button>
  </div>
</div>
`;
